### PR TITLE
Dps updates

### PR DESCRIPTION
**Pull request by wesmc7777**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by wesmc7777** [comment]
#sign-off

File: 101-iothub-device-provisioning/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      "properties": {
+        "iotHubs": [
+          {
+            "connectionString": "[concat('HostName=', reference(resourceId('Microsoft.Devices/Iothubs', parameters('iotHubName'))).hostName, ';SharedAccessKeyName=', variables('iotHubKeyName'), ';SharedAccessKey=', listkeys(resourceId('Microsoft.Devices/Iothubs/Iothubkeys', parameters('iotHubName'), variables('iotHubKeyName')), '2020-03-01').primaryKey)]",
+            "location": "[parameters('location')]",
+            "name": "[reference(resourceId('Microsoft.Devices/Iothubs', parameters('iotHubName'))).hostName]"
+          }
+        ]
+      },
+      "dependsOn": [ "[parameters('iotHubName')]" ]
```
Comment (mumian) [inline 101-iothub-device-provisioning/azuredeploy.json:73]: move "dependsOn" before "properties"

Comment (wesmc7777) [reply 101-iothub-device-provisioning/azuredeploy.json]: Ok, I did. Why does it matter?

Comment (mumian) [reply 101-iothub-device-provisioning/azuredeploy.json]: see "Sort Order of Properties" in https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md.

**Pull request merged**